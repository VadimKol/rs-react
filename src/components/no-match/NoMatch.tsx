import { type ReactNode } from 'react';
import { useNavigate } from 'react-router-dom';

import error_img from '@/assets/images/404.png';

import { CustomButton } from '../custom-button/Ð¡ustomButton';
import styles from './styles.module.scss';

export function NoMatch(): ReactNode {
  const navigate = useNavigate();
  const theme = 'dark';

  return (
    <main className={theme === 'dark' ? 'main' : 'main light'}>
      <img className={styles.error_img} src={error_img} alt="404 error" width={490} height={490} />
      <div className={styles.buttons}>
        <CustomButton className={styles.home} onClick={() => navigate('/')}>
          Home page
        </CustomButton>
      </div>
    </main>
  );
}
