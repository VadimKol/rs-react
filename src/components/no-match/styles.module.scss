.buttons {
  display: flex;
  gap: 30px;
  justify-content: space-between;
}

.error_img {
  display: block;

  @media screen and (width <= 550px) {
    width: 100%;
    height: 100%;
  }
}

.home {
  font-size: 1.3rem;
}
