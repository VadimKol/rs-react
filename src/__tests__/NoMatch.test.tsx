import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { NoMatch } from '@/components/no-match/NoMatch';

describe('NoMatch Component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <NoMatch />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
