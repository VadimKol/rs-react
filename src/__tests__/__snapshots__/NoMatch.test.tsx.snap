// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch Component renders correctly 1`] = `
<div>
  <main
    class="main"
  >
    <img
      alt="404 error"
      class="error_img"
      height="490"
      src="stub-test"
      width="490"
    />
    <div
      class="buttons"
    >
      <button
        class="button home"
        id=""
        type="button"
      >
        Home page
      </button>
    </div>
  </main>
</div>
`;
